Python/Product/PythonTools/PythonTools/Options/PythonDiagnosticsOptionsPage.cs

```diff
@@ -78,13 +78,20 @@ public class PythonDiagnosticsOptionsPage : PythonDialogPage {
         }
 
         private void SaveToFile(bool includeAnalysisLogs) {
-            var path = Site.BrowseForFileSave(
-                _window.Handle,
-                Strings.DiagnosticsWindow_TextFileFilter,
-                PathUtils.GetAbsoluteFilePath(
+            string initialPath = null;
+            try {
+                initialPath = PathUtils.GetAbsoluteFilePath(
                     Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments),
                     Strings.DiagnosticsWindow_DefaultFileName.FormatUI(DateTime.Now)
-                )
+                );
+            } catch (Exception ex) when (!ex.IsCriticalException()) {
+                Debug.Fail(ex.ToUnhandledExceptionMessage(GetType()));
+            }
+
+            var path = Site.BrowseForFileSave(
+                _window?.Handle ?? IntPtr.Zero,
```

Comment:
`_window` can't be null, cause `SaveToFile` is added to `_window.SaveToFile` and this is the only way it can be called.

Reply:
Then this won't ever trigger, and no harm done.